cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -406,15 +408,32 @@ func newDeleteOptions() *metav1.DeleteOptions {
 }
 
 // TODO: Test this function
-func (c *client) UpdateClusterObjectEndpoint(controlPlaneIP, clusterName, namespace string) error {
+func (c *client) UpdateClusterObjectEndpoint(clusterEndpoint, clusterName, namespace string) error {
+	var (
+		endpointHost string
+		endpointPort string
+	)
 	cluster, err := c.GetClusterObject(clusterName, namespace)
 	if err != nil {
 		return err
 	}
+	// clusterEndpoint can be passed as hostname or hostname:port, if port is not present
+	// the default port 443 will be applied.
```

Comment:
Should we have this and a better description in an exported comment?

Reply:
took a stab at it